src/util.rs

```diff
@@ -135,7 +141,14 @@ pub struct TimeMacroFinder {
     pub found_time: bool,
     pub found_timestamp: bool,
     overlap_buffer: [u8; MAX_HAYSTACK_LEN * 2],
-    counter: usize,
+    /// Counter of chunks of full size we've been through. Partial reads do
+    /// not count and are handled separately.
+    full_chunks_counter: usize,
+    /// Contents of the previous read if it was smaller than `MAX_HAYSTACK_LEN`,
+    /// plus MAX_HAYSTACK_LEN bytes of the previous chunk, to account for
+    /// the possibility of partial reads splitting a time macro
+    /// across two calls.
+    previous_small_read: Option<Vec<u8>>,
```

Comment:
These changes should go in the previous commit rather than this one. I wonder if it wouldn't be simpler to use a single buffer instead of overlap_buffer + previous_small_read. (I'll also note an `Option<Vec<T>>` is kind of pointless if you don't expect there to be a Some(vec![]).

Reply:
> These changes should go in the previous commit rather than this one.

Good catch.

> I wonder if it wouldn't be simpler to use a single buffer instead of overlap_buffer + previous_small_read.

IMO it would make the logic harder to read.

> (I'll also note an `Option<Vec<T>>` is kind of pointless if you don't expect there to be a Some(vec![]).

True. While it doesn't matter in terms of performance, I'll switch to just a `Vec<u8>`, should make the code simpler.